src/server.ts

```diff
@@ -218,90 +189,87 @@ const getCAmsg = (deps, diagnostics, totalCount): string => {
 
 const caDefaultMsg = 'Checking for security vulnerabilities ...';
 
-const metadataCache = new Map();
-const get_metadata = (ecosystem, name, version) => {
-    return new Promise((resolve, reject) => {
-        const cacheKey = ecosystem + " " + name + " " + version;
-        const metadata = metadataCache[cacheKey];
-        if (metadata != null) {
-            logger.info('cache hit for ' + cacheKey);
-            connection.console.log('cache hit for ' + cacheKey);
-            resolve(metadata);
+/* Fetch Vulnerabilities by component-analysis batch api-call */
+const fetchVulnerabilities = async (reqData) => {
+    try {

```

Comment:
Add try exactly to the statement which would raise Exception. In this case, it would be `fetch` and `response.json()`.